mockup/patterns/tinymce/js/links.js

```diff
@@ -582,41 +600,42 @@ define([
         }
       }
     },
+
     guessImageLink: function(src) {
-      var self = this;
-      if (src.indexOf(self.options.prependToScalePart) !== -1) {
-        self.linkType = 'image';
-        self.$scale.val(self.tinypattern.getScaleFromUrl(src));
-        self.linkTypes.image.set(self.tinypattern.stripGeneratedUrl(src));
+      if (src.indexOf(this.options.prependToScalePart) !== -1) {
+        this.linkType = 'image';
+        this.$scale.val(this.tinypattern.getScaleFromUrl(src));
+        this.linkTypes.image.set(this.tinypattern.stripGeneratedUrl(src));
       } else {
-        self.linkType = 'externalImage';
-        self.linkTypes.externalImage.set(src);
+        this.linkType = 'externalImage';
+        this.linkTypes.externalImage.set(src);
       }
     },
+
     guessAnchorLink: function(href) {
-      var self = this;
```

Comment:
To be honest, I don't particularly like it. It looks to me like a Python developer trying to write Javascript.

Generally, the consensus in the Javascript community appears to be to use ``that`` as variable name if you want to refer to the context enclosing a callback (i.e instead of ``self``). This is what Douglas Crockford suggests in "Javascript, the good parts".

In general however, I prefer using things like ``function.prototype.bind`` or ``$.proxy`` to explicitly set the context of ``this`` for callback functions.